client/API.md

```diff
@@ -0,0 +1,536 @@
+
+| HTTP Response Code | Meaning                                                                                                                                                             |
+| ------------------ | -------                                                                                                                                                             |
+| 200                | The request was a success the content is dependent on the request.                                                                                                  |
+| 204                | The request was a success and no content was returned.                                                                                                              |
+| 4xx                | Invalid request, refer to error for what it wrong with the request. Repeating the request will continue to return the same error.                                   |
+| 5xx                | The server was unable to process the request, refer to the error for a reason. Repeating the request may result in a success if the server issue has been resolved. |
+
+### Errors
+
+All requests can return JSON in the following format to provide more information about a failed request.
+
+```
+{
+    "error" : "error message"
+}
+```
+
+### Query Parameters vs JSON body
+
+To make using this API a consistent and easy experience we follow one simple rule for when extra information
+about a request is found in the query parameters of the URL or when they are part of the submitted JSON body.
+
+Query parameters are used only for GET requests and all POST requests expect parameters to be specified in the JSON body.
+
+## Tasks
+
+A task represents work for Kapacitor to perform.
+A task is defined by its name, type, TICKscript, and list of database retention policy pairs it is allowed to access.
+
+### Define Task
+
+To define a task POST to the `/task/TASK_NAME` endpoint.
+If no task exists it with that name it will be created.
+Otherwise the existing task will be updated with the provided information.
+
+Define a task using a JSON object with the following options:
+
+| Parameter | Purpose                                                                |
+| --------- | -------                                                                |
+| type      | The task type: `stream` or `batch`.                                    |
+| dbrps     | List of database retention policy pairs the task is allowed to access. |
+| script    | The content of the script.                                             |
+
+If any option is missing it will be left unmodified if the task already exists.
+
+#### Example
+
+```
+POST /task/TASK_NAME
+{
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream\n    |from()\n        .measurement('cpu')\n"
+}
+```
+
+Modify only the dbrps of the task.
+
+```
+POST /task/TASK_NAME
```

Comment:
Modifying an existing REST resource should be PATH instead of POST.  Many times it is important to not create an entry more than once.